docker/l2s/src/index.ts

```diff
@@ -0,0 +1,232 @@
+  const tokenPriceQuery = `SELECT h.id as id, h.timestamp as timestamp, close as price,
+                  a.eth as address, t.decimals as decimals
+                  FROM cmc_historical h
+                  JOIN cmc_address a ON h.id = a.id
+                  JOIN cmc_eth_token t on t.address = a.eth
+                  WHERE a.eth IN ({addresses: Array(TINYTEXT)})
+                  AND toDate(h.timestamp) = toDate({date: date})
+                  LIMIT 1000`;
+  const tokenPriceResult = await client.query({
+    query: tokenPriceQuery,
+    query_params: {
+      addresses: tokenAddresses,
+      date: dateString,
+    },
+    format: "JSONEachRow",
+  });
+  const tokenPrices: TokenPrice[] = await tokenPriceResult.json();
+  console.log("Get erc20 tokens historical price data from ClickHouse success");
+
+  const ethPriceQuery = `SELECT id, timestamp, close as price,'NATIVE_TOKEN' as address, 18 as decimals
+                          FROM cmc_historical
+                          WHERE id = {id: Int64}
+                          AND toDate(timestamp) = toDate({date: date})
+                          LIMIT 1`;
+  const ethPriceResult = await client.query({
+    query: ethPriceQuery,
+    query_params: {
+      id: CMC_ETH_ID,
+      date: dateString,
+    },
+    format: "JSONEachRow",
+  });
+  const ethPrices: TokenPrice[] = await ethPriceResult.json();
+  console.log("Get eth historical price data from ClickHouse success");
+
+  // Close clickhouse client connection
+  await client.close();
+
+  const decimalMap = tokenPrices.reduce((acc, obj) => {
+    acc[obj.address] = obj.decimals;
+    return acc;
+  }, {});
+
+  const priceMap = tokenPrices.reduce((acc, obj) => {
+    acc[obj.address] = obj.price;
+    return acc;
+  }, {});
+
+  const priceTimestamp = ethPrices[0].timestamp;
+
+  const escrowContractTokens: any[] = escrowContracts.map((obj) => ({
+    ...obj,
+    decimals: decimalMap[obj.token_address.toLowerCase()],
+    price: priceMap[obj.token_address.toLowerCase()],
+  }));
+
+  const ethEscrowContractTokens: any[] = ethEscrowContracts.map((obj) => ({
+    ...obj,
+    decimals: DEFAULT_DECIMAL,
+    price: ethPrices[0].price,
+  }));
+
+  //  Initialize provider
+  const rpcProvider = new StaticJsonRpcProvider(config.rpcUrl);
+  const multicallContract = new Contract(
+    config.multicallContractAddress,
+    multicallABI,
+    rpcProvider
+  );
+  const CONTRACT_FUNCTION = "balanceOf";
+  const contracts: Contract[] = [];
+  const callInputs: string[][] = [];
+  const ethEscrow: string[] = [];
+
+  escrowContractTokens.forEach((ctr) => {
+    contracts.push(new Contract(ctr.token_address, erc20ABI, rpcProvider));
+    callInputs.push([ctr.address]);
+  });
+  ethEscrowContractTokens.forEach((ctr) => {
+    ethEscrow.push(ctr.address);
+  });
+
+  console.log("Start getting tokens balance by multicall contract");
+  const tx = await getMultipleContractMultipleData(
+    contracts,
```

Comment:
@viettiennguyen029  What is the maximum number of parameters in one multical (1000/2000/5000)? Should limit the number of parameters in one multicall (100/200/500)?